gamemodes/terrortown/gamemode/client/cl_main.lua

```diff
@@ -456,8 +456,14 @@ local function RoundStateChange(o, n)
         -- clear decals in cache from previous round
         util.ClearDecals()
 
-        -- resets bone positions that fixes broken fingers on bad addons
-        weaponrenderer.ResetBonePositions(LocalPlayer():GetViewModel())
+        local client = LocalPlayer()
+
+        -- Resets bone positions that fixes broken fingers on bad addons.
+        -- When late-joining a server this function is executed before the local player
+        -- is completely set up. Therefore we safeguard this with this check.
+        if IsValid(client) and isfunction(client.GetViewModel) then
+            weaponrenderer.ResetBonePositions(client:GetViewModel())
+        end
```

Comment:
This would mean that a late-joining might have broken bone positions until the roundstate changes right?

It should fix the immediate issue, but we really need to find out how to prevent these late-join issues in general.

Reply:
> This would mean that a late-joining might have broken bone positions until the roundstate changes right?

No as these broken bones only happen when bad addons break them. Strictly speaking this line isn't needed in TTT2, it just adds a protection for addons such as the harpoon that break bones.

Reply:
in addition to doing broken bone position checks, an enhancement would be to reset the alphas as well, sometimes the slide on the deagle or everything but the buttons on C4 get disappeared as well -- anything bad that could possibly happen to a viewmodel part probably will happen

resetting everything between rounds is pretty inconsequential as well

Reply:
The buttons disappearing could be due to broken bones as well. It might sound illogical, but the C4 has bones too. The "classic" SWEP Construction Toolkit weapons hide the view model by modifying its bones so it is no longer visible. When this is done, this can lead to artifacts such as the buttons not showing on the C4